executors/cmdexec/block_volume.go

```diff
@@ -192,3 +194,163 @@ func (c *CmdExecutor) ListBlockVolumes(host string, blockhostingvolume string) (
+		output = results[0].ErrOutput
+	}
+
+	var blockVolumeInfoExec CliOutput
+	err = json.Unmarshal([]byte(output), &blockVolumeInfoExec)
+	if err != nil {
+		logger.Warning("Unable to parse gluster-block info output [%v]: %v",
+			output, err)
+		err = fmt.Errorf("Unparsable error during block volume info: %v",
+			output)
+	}
+
+	// if any of the cases above set err, log it and return
+	if err != nil {
+		logger.LogError("%v", err)
+		return nil, err
+	}
+
+	var blockVolumeInfo executors.BlockVolumeInfo
+
+	blockVolumeInfo.BlockHosts = blockVolumeInfoExec.Portal
+	blockVolumeInfo.GlusterNode = host
+	blockVolumeInfo.GlusterVolumeName = blockVolumeInfoExec.BlockHostingVolumeName
+	blockVolumeInfo.Hacount = blockVolumeInfoExec.Ha
+	blockVolumeInfo.Iqn = "iqn.2016-12.org.gluster-block:" + blockVolumeInfoExec.Gbid
+	blockVolumeInfo.Name = blockVolumeInfoExec.BlockName
+	blockVolumeInfo.Size, err = ToGigaBytes(blockVolumeInfoExec.Size)
+	if err != nil {
+		logger.LogError("Error: %v : %v", err, blockVolumeInfoExec.Size)
+		return nil, err
+	}
+	blockVolumeInfo.Username = blockVolumeInfoExec.Gbid
+	blockVolumeInfo.Password = blockVolumeInfoExec.Password
+
+	return &blockVolumeInfo, nil
+}
+
+func (s *CmdExecutor) BlockVolumeExpand(host string, blockHostingVolumeName string, blockVolumeName string, newSize int) error {
+	godbc.Require(host != "")
+	godbc.Require(blockHostingVolumeName != "")
+	godbc.Require(blockVolumeName != "")
+	godbc.Require(newSize > 0)
+
+	commands := []string{
+		fmt.Sprintf("gluster-block modify %v/%v  size %vGiB --json",
+			blockHostingVolumeName, blockVolumeName, newSize),
+	}
+	results, err := s.RemoteExecutor.ExecCommands(host, rex.ToCmds(commands), 10)
+	if err != nil {
+		// non-command error conditions
+		return err
+	}
+
+	output := results[0].Output
+	if output == "" {
+		output = results[0].ErrOutput
+	}
+
+	type CliOutput struct {
+		Iqn       string   `json:"IQN"`
+		Size      string   `json:"SIZE"`
+		SuccessOn []string `json:"SUCCESSFUL ON"`
+		FailedOn  []string `json:"FAILED ON"`
+		SkippedOn []string `json:"SKIPPED ON"`
+		Result    string   `json:"RESULT"`
+		ErrCod    int      `json:"errCode"`
+		ErrMsg    string   `json:"errMsg"`
+	}
+
+	var blockVolumeExpand CliOutput
+	err = json.Unmarshal([]byte(output), &blockVolumeExpand)
+	if err != nil {
+		logger.Warning("Unable to parse gluster-block output [%v]: %v",
+			output, err)
+		err = fmt.Errorf("Unparsable error during block volume expand: %v",
+			output)
+	} else if blockVolumeExpand.Result == "FAIL" {
+		// the fail flag was set in the output json
+		err = fmt.Errorf("Failed to expand block volume: %v",
+			blockVolumeExpand.ErrMsg)
+	} else if !results.Ok() {
+		// the fail flag is not set but the command still
+		// exited non-zero for some reason
+		err = fmt.Errorf("Failed to expand block volume: %v",
+			results[0].Error())
+	}
+
+	// if any of the cases above set err, log it and return
+	if err != nil {
+		logger.LogError("%v", err)
```

Comment:
At least give some context to the person who might grep the code for an error line seen in the logs.